src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -193,6 +198,64 @@ public void deleteRoles(final Node node) throws RepositoryException {
     };
 
     /**
+     *  This maps a Collection of acl:agentClass values to a List of agents.
+     *  Any out-of-domain URIs are silently ignored.
+     */
+    private List<String> dereferenceAgentClasses(final Collection<String> agentClasses) {
+        final Session internalSession = sessionFactory.getInternalSession();
+        final IdentifierConverter<Resource, FedoraResource> translator =
+                new DefaultIdentifierTranslator(internalSession);
+
+        final List<String> members = new ArrayList<>();
+        agentClasses.stream()
+                    .distinct()
+                    .forEach(x -> {
+                        if (x.startsWith(FEDORA_INTERNAL_PREFIX)) {
+                            final FedoraResource resource = nodeService.find(
```

Comment:
The `rdfs:Domain` of `foaf:member` is `foaf:Group`. I am not currently enforcing that, but I'm beginning to think that I should. I can imagine a scenario in which various vocabularies are supported for defining membership (e.g. ore:aggregates), and so checking the type could be very useful.